File: tests/bench.py
import RPi.GPIO as GPIO
import time

pin = 11
GPIO.setmode(GPIO.BOARD)
GPIO.setup(pin,GPIO.IN,GPIO.PUD_DOWN)

t = time.time()
for i in range(0,100000,1):
    #pass
    GPIO.input(pin)
print( "%f"%(time.time() - t ) )
exit()

File: tests/bench.php
#!/usr/bin/env php
<?php
	$pin = 11;
	$wiringpi = new WiringPi();
	$wiringpi->setmode(WiringPi::WPI_MODE_PHYS);
	$wiringpi->setup( $pin, WiringPi::INPUT, WiringPi::PUD_OFF );

	$start = microtime( true );
	for ( $n = 0; $n < 100000; $n ++ ) {
		$wiringpi->input( $pin );
	}
	$end = microtime( true );
//	echo "start: ${start}, end: ${end}\n";
	echo ( $end - $start ) . "\n";
?>
